packages/legend-application-studio/src/components/editor/edit-panel/project-configuration-editor/ProjectConfigurationEditor.tsx

```diff
@@ -450,6 +431,131 @@ const ProjectDependencyEditor = observer(
+    });
+
+    const isNotValidPlatformVersion = action(
+      (val: string): boolean => val === '',
+    );
+
+    return (
+      <PanelFormTextEditorWithValidation
+        isReadOnly={isReadOnly}
+        value={platform.platformVersion}
+        name={platform.name}
+        update={changeVersion}
+        validationErrorMessage={
+          isNotValidPlatformVersion(platform.platformVersion)
+            ? 'Version cannot be empty'
+            : undefined
+        }
+      />
+    );
+  },
+);
+
+const ProjectPlatformVersionEditor = observer(
+  (props: { projectConfig: ProjectConfiguration; isReadOnly: boolean }) => {
+    const { projectConfig, isReadOnly } = props;
+    const editorStore = useEditorStore();
+
+    const platformConfigurations = projectConfig.platformConfigurations;
+
+    if (platformConfigurations) {
+      editorStore.projectConfigurationEditorState.setOverrideDefaultPlatform(
+        true,
+      );
+    }
+
+    const isEditingPlatformConfigs =
+      editorStore.projectConfigurationEditorState.overridesDefaultPlatform;
+
+    const defaultPlatforms = editorStore.sdlcServerClient.platforms;
+
+    const overridePlatforms = (): void => {
+      if (isEditingPlatformConfigs !== true) {
+        editorStore.setActionAlertInfo({
+          message:
+            'This is an advanced configuration option meant for users who already have in mind the platform version they would like to input.',
+          prompt:
+            'Do you still want to proceed to override default project configurations?',
+          type: ActionAlertType.CAUTION,
+          onEnter: (): void => editorStore.setBlockGlobalHotkeys(true),
+          onClose: (): void => editorStore.setBlockGlobalHotkeys(false),
+          actions: [
+            {
+              label: 'Proceed',
+              type: ActionAlertActionType.PROCEED_WITH_CAUTION,
+              handler: (): void => {
+                editorStore.projectConfigurationEditorState.setOverrideDefaultPlatform(
+                  !isEditingPlatformConfigs,
+                );
+              },
+            },
+            {
+              label: 'Cancel',
+              type: ActionAlertActionType.PROCEED,
+              default: true,
+            },
+          ],
+        });
+      } else {
+        editorStore.projectConfigurationEditorState.setOverrideDefaultPlatform(
+          !isEditingPlatformConfigs,
+        );
+      }
+    };
+
+    return (
+      <Panel>
+        <PanelForm>
+          <PanelSection>
+            <PanelFormBooleanEditor
+              value={isEditingPlatformConfigs}
+              name=""
+              description="Override default platform configurations"
+              isReadOnly={isReadOnly}
+              update={overridePlatforms}
+            />
+
+            {defaultPlatforms &&
+              !isEditingPlatformConfigs &&
+              defaultPlatforms.map((p) => (
+                <PlatformBasicEditor
+                  key={p.name}
```

Comment:
tricky thing here is we want to preserve the casing of the platform configuration name, `legend-engine` should be left as such and not `Legend engine`

Reply:
adding `preserveCasing` optional prop in component 🤔